libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationPerformanceEvent.java

```diff
@@ -19,20 +21,36 @@
   private static final SimpleDateFormat dateFormat = new SimpleDateFormat(DATE_AND_TIME_PATTERN, Locale.US);
   private static final String PERFORMANCE_TRACE = "mobile.performance_trace";
   private static final String EVENT_NAME = "event_name";
+  private static final String SCREEN_DENSITY = "screen_density";
 
   private final String event;
   private final String created;
   private final String sessionId;
   private final List<Counter> counters;
   private final List<Attribute> attributes;
+  NavigationPerformanceMetadata metadata;
 
-  NavigationPerformanceEvent(String sessionId, String eventName) {
+  NavigationPerformanceEvent(String sessionId, String eventName,
+                             NavigationPerformanceMetadata metadata) {
+    this(null, sessionId, eventName, metadata);
+  }
+
+  NavigationPerformanceEvent(@Nullable Context context, String sessionId, String eventName,
+                             NavigationPerformanceMetadata metadata) {
     this.event = PERFORMANCE_TRACE;
     this.created = obtainCurrentDate();
     this.sessionId = sessionId;
     this.counters = new ArrayList<>();
     this.attributes = new ArrayList<>();
+    this.metadata = metadata;
     attributes.add(new Attribute(EVENT_NAME, eventName));
+    if (context != null) {
```

Comment:
Is this necessary? Wondering if we always want to capture the screen density 🤔 I guess so, right?